examples/frame_formats.proto

```diff
@@ -0,0 +1,840 @@
+// Frame Format Definitions for struct-frame
+// 
+// This file defines various message framing formats that can be used for
+// reliable communication over serial links, network sockets, or any byte stream.
+//
+// Each frame format provides different tradeoffs between:
+// - Overhead (header/footer bytes)
+// - Error detection (CRC/checksum)
+// - Message boundary detection
+// - Length-prefixed vs fixed-length messages
+//
+// Use Case Summary:
+// ================
+// NoFormat: Direct point-to-point trusted links, shared memory, nested protocols
+// MsgIdFrame: Minimal framing with error detection
+// MsgIdFrameNoCrc: Minimal framing for trusted links
+// BasicFrame: Standard reliable communication with sync recovery
+// BasicFrameNoCrc: Reduced overhead when CRC handled elsewhere
+// TinyFrame: Low overhead for constrained environments
+// TinyFrameNoCrc: Minimal overhead for trusted constrained links
+// *WithLength variants: Variable length messages, streaming data
+// UBX: u-blox GPS receiver compatibility
+// MavlinkV1: ArduPilot/PX4 legacy drone communication
+// MavlinkV2: Modern ArduPilot/PX4 drone communication with authentication
+
+package frame_formats;
+
+// ============================================================================
+// Frame Format Enumeration
+// ============================================================================
+
```

Comment:
Create a BasicMessage message that will be included in every single frame format.